### PR TITLE
[Vulnerabilities dashboards] Add search bar services

### LINKED ISSUE #5945: [Vulnerabilities dashboards] Add search bar services/hooks
## Description

All services to avoid the use of the SearchBar component from OSD data plugin.
These services will be used like a search bar in the new vulnerabilities dashboards.

## Tasks

- [x] Create services/hooks
- [x] Unit test for every services/hooks

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### PR DESCRIPTION
**Pull request by Machi3mfl**
### Description
Add search bar services

Closes #5945

## Unit test added

-  should return default app index pattern when not receiving a default index pattern
-  should return the same index pattern when receiving a default index pattern
-  should show an ERROR message and get the default app index pattern when not found the index pattern data by the ID received
-  should return the same filters and apply them to the filter manager when are received by props
-  should return and preserve filters when the index pattern received is equal to the index pattern already selected in the app 
-  should return empty filters when the index pattern is NOT equal to the default app index pattern

Screenshot 2023-09-29 at 16 57 07

### Check List
- [x] All tests pass
  - [x] `yarn test:jest`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by Machi3mfl** [comment]
## Use Example

```tsx
  const SearchBar = getPlugins().data.ui.SearchBar;

  const { searchBarProps } = useSearchBar({
    defaultIndexPatternID: VULNERABILITIES_INDEX_PATTERN_ID,
    onFiltersUpdated: (filters) => {
      console.log(filters);
    },
    onQuerySubmitted: (payload) => {
      console.log(payload);
    },
  });

  return (
    <>
      <I18nProvider> {/* The searchbar is not rendered when is not wrapped by the i18n provider */}
        <SearchBar
          {...searchBarProps}
        />
      </I18nProvider>
    </>
  );

```

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+ */
+const useSearchBarConfiguration = (props?: tUseSearchBarProps): tUserSearchBarResponse => {
+    // dependencies
+    const filterManager = useFilterManager().filterManager as FilterManager;
+    const [query, setQuery] = props?.query ? useState(props?.query) : useQueryManager();
+    const { timeFilter, timeHistory, setTimeFilter } = useTimeFilter();
+    // states
+    const [isLoading, setIsLoading] = useState(false);
+    const [indexPatternSelected, setIndexPatternSelected] = useState<IIndexPattern>();
+    const [error, setError] = useState();
```
Comment (lucianogorza) [inline plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx:35]: Unused code
```suggestion
```

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    /**
+     * Search bar properties necessary to render and initialize the osd search bar component
+     */
+    const searchBarProps: Partial<SearchBarProps> = {
+        isLoading,
+        ...indexPatternSelected && { indexPatterns: [indexPatternSelected] }, // indexPattern cannot be empty or empty []
+        filters: filterManager.getFilters(),
+        query: query,
```
Comment (lucianogorza) [inline plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx:105]:
```suggestion
        query,
```

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        timeHistory: timeHistory,
```
Comment (lucianogorza) [inline plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx:106]:
```suggestion
        timeHistory,
```

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-search-bar-configuration.test.ts [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+import '@testing-library/jest-dom/extend-expect';
+// osd dependencies
+import { Start, dataPluginMock } from '../../../../../../../src/plugins/data/public/mocks';
+import {
+  Filter,
+  IndexPattern,
+  Query,
+  TimeRange,
+} from '../../../../../../../src/plugins/data/public';
+import { createFilterManagerMock } from '../../../../../../../src/plugins/data/public/query/filter_manager/filter_manager.mock';
```
Comment (lucianogorza) [inline plugins/main/public/components/overview/vulnerabilities/searchbar/use-search-bar-configuration.test.ts:11]: Unused code
```suggestion
```

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+     * @returns 
+     */
+    const getIndexPattern = async (indexPatternID?: string) => {
+        const indexPatternService = getDataPlugin().indexPatterns as IndexPatternsContract;
+        if(indexPatternID){
+            try{
+                return await indexPatternService.get(indexPatternID);
+            }catch(error){
+                // when the index pattern id not exists will get the default
+                console.error(error);
```
Comment (lucianogorza) [inline plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx:65]: It's ok to log the error or it's common behavior?

**Review by lucianogorza — CHANGES_REQUESTED** [review]

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 1 above)
Comment (jbiset) [inline plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx]: This state is not being used. Right?

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 2 above)
Comment (jbiset) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx]: Idem

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 3 above)
Comment (jbiset) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx]: Idem

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+        onFiltersUpdated: (filters: Filter[]) => {
+            // its necessary execute setter to apply filters
+            filterManager.setFilters(filters);
+            props?.onFiltersUpdated && props?.onFiltersUpdated(filters);
+        },
+        onQuerySubmit: (payload: { dateRange: TimeRange; query?: Query }, _isUpdate?: boolean): void => {
+            const { dateRange, query } = payload;
+            // its necessary execute setter to apply query filters
+            setTimeFilter(dateRange);
+            setQuery(query)
```
Comment (jbiset) [inline plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx:117]: Missing `;`

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 5 above)
Comment (Machi3mfl) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx:65]: It's a common behavior from osd indexPatternService but I want to leave a message to the developer. This case is when the hook receives an id but doesn't exist. This message can help to understand why the index pattern is not selected in the search bar

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx [hunk 7]
```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState } from 'react'
+import { SearchBarProps, FilterManager, TimeRange, Query } from '../../../../../../../src/plugins/data/public'
+import { Filter, IIndexPattern, IndexPatternsContract } from '../../../../../../../src/plugins/data/public';
+import { getPlugins, getDataPlugin } from '../../../../kibana-services';
```
Comment (jbiset) [inline plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx:4]: `getPlugins` is defined but never used

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-search-bar-configuration.test.ts (hunk 4 above)
Comment (Machi3mfl) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-search-bar-configuration.test.ts]: Resolved in https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/2348731fb1410f74abdaeee6920585942d2e2c0a

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 3 above)
Comment (Machi3mfl) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx]: Resolved in https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/2348731fb1410f74abdaeee6920585942d2e2c0a

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 2 above)
Comment (Machi3mfl) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx]: Resolved in https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/2348731fb1410f74abdaeee6920585942d2e2c0a

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 1 above)
Comment (Machi3mfl) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx]: Resolved in https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/2348731fb1410f74abdaeee6920585942d2e2c0a

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx
Comment (jbiset) [inline plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx:1]: Fix file name `use-seach-bar-configuration.tsx` to `use-search-bar-configuration.tsx`

**Review by jbiset — CHANGES_REQUESTED** [review]
## CR: :yellow_circle: (Commented)

## Test :green_circle: 

Run_test

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 1 above)
Comment (Machi3mfl) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx]: Yes, fixed in https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/2348731fb1410f74abdaeee6920585942d2e2c0a

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 7 above)
Comment (Machi3mfl) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx]: Fixed in [0c834a5](https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/0c834a5400a421c1799ec97fd6645492540346af)

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx
Comment (Machi3mfl) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx:1]: Fixed in [0c834a5](https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/0c834a5400a421c1799ec97fd6645492540346af)

File: plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx (hunk 6 above)
Comment (Machi3mfl) [reply plugins/main/public/components/overview/vulnerabilities/searchbar/use-seach-bar-configuration.tsx]: Fixed in [0c834a5](https://github.com/wazuh/wazuh-kibana-app/pull/5960/commits/0c834a5400a421c1799ec97fd6645492540346af)

**Review by jbiset — APPROVED** [review]

**Comment by lucianogorza** [comment]
## CR: :green_circle:

## Test :green_circle: 

image

**Review by lucianogorza — APPROVED** [review]

**Pull request merged**